File: lab2/q1starter.cpp
#include <string>
#include <iostream>

auto main() -> int
{
	std::string lineInput;
	int numRead = 0;
	int totalLength = 0;
	float avgLength = 0;

	while ( getline( std::cin, lineInput ) )
	{
		int lineLength = lineInput.length();

		numRead = numRead + 1; // increment 

		auto stringLineLength = std::to_string( lineLength );

		totalLength = totalLength + lineLength; // increment 
		
		auto outputLine = stringLineLength + "\t" + lineInput;
  		std::cout << outputLine << std::endl;
	}

	// convert to float 
	float totalLengthFloat = static_cast<float>(totalLength);
	float linesReadFloat = static_cast<float>(numRead);

	// find avg length of lines 
	avgLength = totalLengthFloat/linesReadFloat;

	// convert everyhting to string 
	std::string linesString = std::to_string(numRead);
	std::string totalLengthString = std::to_string(totalLength);
	std::string averageLengthString = std::to_string(avgLength);

	// output footer 
	std::cout << "lines: " << linesString << "\t" << "total length: " << totalLengthString << "\t" << "average length: " << averageLengthString << std::endl;
	return 0;
}

File: lab2/q1starter.py
import sys

num_read = 0
total_lines = 0
avg_length = 0

for current_line in sys.stdin:
	num_read += 1 #increment 

	strip_line = current_line.strip()
	strip_length = len( strip_line )

	total_lines += strip_length #add curent length to total legnth 
	length_as_string = str( strip_length )
	output_string = length_as_string + "\t" + current_line
	
	print( output_string, end='' )

# find average length of line 
avg_length = total_lines/num_read

# convert all variables to strings 
lines_string = str(num_read)
total_lengths_string = str(total_lines)
average_length_string = str(avg_length)

# output footer 
print ("lines: " + lines_string + "\t" + "total length: " + total_lengths_string + "\t" + "average length: " + average_length_string)
